### ISSUE TITLE
suppress-notifications results in empty [Placeholder Name] work items

### ISSUE THREAD

**Issue by canadaleks** [solidify/jira-azuredevops-migrator#965]
**Describe the problem**

When performing a `wi-import` utilizing a configuration json derived from the `config-agile.json` sample, the imported work items in Azure DevOps all have the title of `[Placeholder Title]` and do not contain any data other than the original work item creation (the iteration path, area path mapped in configuration are also not captured).

Setting logging to Debug, there's no indication of an issue in the import process.

When reverting `suppress-notifications` to `false`, the import works as expected.

**To Reproduce**

1. (assuming a ready-to-run mapping config json based on config-agile sample file is present) Set the `suppress-notifications` option in your mapping json to `true`
2. Run the `.\wi-import --token TOKEN_HERE--url URL_HERE --config config-agile-test.json --force`
3. Run query in Azure DevOps to list newly created work items

**Tool version**
v3.0.335

**Attachments**

Please attach the following files:
- [config-agile-claims.json](https://github.com/solidify/jira-azuredevops-migrator/files/13908598/config-agile-claims.json)

**Comment by Alexander-Hjelm** [comment]
Check the import log and verify that there are no errors or warnings that need addressing.

Usually this problem is indicating that there are underlying configuration errors. But you need to get the specific error message in the logs. These should have further instructions on what you need to do.

**Comment by canadaleks** [comment]
I ran this export and import again (with a smaller subset of Jira issues to get a smaller log file).

For the export logfile, there are no Error messages, and the only warning messages are about not finding users in the user mapping file (expected).

For the import logfile, there are a couple of errors, which are specifically due to a Jira issue referencing attachments which are no longer available, thus resulting in a 'Value cannot be null' error (listed below) when trying to create the attachment in Azure DevOps.  However, this is for one specific Jira issue being imported (out of the two that were imported in this smaller run).  The two imported tickets have a title of [Placeholder Text] and the ticket is completely empty (not area path is not assigned properly either).

When running the import, I had Fiddler running to I could capture what info was being sent to ADO, and all the PATCH calls to update the work item have [Placeholder Name] for the Title and all the other fields I mapped it never sets the proper value on any of the revisions processed.  

When I set suppressNotifications back to false and run the import, everything works fine (but spams people with emails).

----

Attachment error:

[I][13:16:10] Adding single attachment
[I][13:16:10] ID: 8a69935a-41be-4ac7-9430-73eabb333d0e
[I][13:16:10] URL: '################################'
[I][13:16:10] 
[E][13:16:11] Failed to import revisions for '1086'.
[System.ArgumentNullException] System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
   at WorkItemImport.WitClientUtils.AddSingleAttachmentToWorkItemAndSave(WiAttachment att, WorkItem wi, Settings settings, Nullable`1 changedDate, String changedBy)
   at WorkItemImport.WitClientUtils.SaveWorkItemAttachments(WiRevision rev, WorkItem wi, Settings settings)
   at WorkItemImport.Agent.ImportRevision(WiRevision rev, WorkItem wi, Settings settings): 
   at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)
   at WorkItemImport.WitClientUtils.AddSingleAttachmentToWorkItemAndSave(WiAttachment att, WorkItem wi, Settings settings, Nullable`1 changedDate, String changedBy)
   at WorkItemImport.WitClientUtils.SaveWorkItemAttachments(WiRevision rev, WorkItem wi, Settings settings)
   at WorkItemImport.Agent.ImportRevision(WiRevision rev, WorkItem wi, Settings settings)

**Comment by Alexander-Hjelm** [comment]
Looks like something is out of order here. I will try and replicate the problem on our end asap.

**Comment by dmazuronak** [comment]
I can confirm this. The error occurs because the _witClientWrapper.UpdateWorkItem function in line 876 of WitClientUtils.cs returns a null value when notification is disabled.

**Comment by canadaleks** [comment]
I have identified the issue.  [Line 107](https://github.com/solidify/jira-azuredevops-migrator/blob/8759bd17805f0ddacfd41af44270081bebcd1bef/src/WorkItemMigrator/WorkItemImport/WitClient/WitClientWrapper.cs#L107) in `WitClientWrapper.cs` is not passing the `suppressNotification` setting properly in `UpdateWorkItemAsync' which looks to be then causing all the revision updates to blank out:

Currently:
```return WitClient.UpdateWorkItemAsync(document: patchDocument, id: workItemId, suppressNotifications, bypassRules: true, expand: WorkItemExpand.All).Result;```

Should be:
```return WitClient.UpdateWorkItemAsync(document: patchDocument, id: workItemId, suppressNotifications: suppressNotifications, bypassRules: true, expand: WorkItemExpand.All).Result;```

I updated this locally and confirmed the issue is now gone.  I tried to create a new branch off master to open a PR for this but it seems that's locked down, so just adding the info here.  Hope this helps!

Thanks.
